inference-engine/tests/unit/cpu/shape_inference_test/assign_shape_inference.cpp

```diff
@@ -0,0 +1,50 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <gtest/gtest.h>
+
+#include <assign_shape_inference.hpp>
+#include <openvino/op/ops.hpp>
+#include <openvino/op/parameter.hpp>
+#include "utils/shape_inference/static_shape.hpp"
+
+using namespace ov;
+template <class T>
+std::shared_ptr<T> constructGraph();
+
+template <>
+std::shared_ptr<op::v3::Assign> constructGraph() {
+    auto input = std::make_shared<op::v0::Parameter>(element::f32, PartialShape{-1, -1, -1, -1});
+    auto read_value = std::make_shared<op::v3::ReadValue>(input, "variable_id");
+    return std::make_shared<op::v3::Assign>(read_value, "variable_id");
+}
+
+template <>
+std::shared_ptr<op::v6::Assign> constructGraph() {
+    auto input = std::make_shared<op::v0::Parameter>(element::f32, PartialShape{-1, -1, -1, -1});
+    auto variable = std::make_shared<ov::op::util::Variable>(
+        ov::op::util::VariableInfo{PartialShape::dynamic(), element::dynamic, "ID"});
+    auto read_value = std::make_shared<op::v6::Assign>(input, variable);
+    return std::make_shared<op::v6::Assign>(read_value, variable);
+}
+
+template <class T>
+void assignTest() {
+    auto assign = constructGraph<T>();
+    // Test PartialShape
+    std::vector<PartialShape> input_shapes = {PartialShape{1, 2, 64, 64}}, output_shapes = {PartialShape{}};
+    shape_infer(assign.get(), input_shapes, output_shapes);
```

Comment:
Please, use `shape_inference` function in all cpu unit tests
```suggestion
    shape_inference(assign.get(), input_shapes, output_shapes);
```